<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>841</width>
    <height>691</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Keyshare</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/logo/logo_st</normaloff>:/logo/logo_st</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="receiveModeButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Manjari</family>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(198, 212, 220);</string>
        </property>
        <property name="text">
         <string>&amp;Receive</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="sendModeButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Manjari</family>
          <pointsize>14</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(198, 212, 220);</string>
        </property>
        <property name="text">
         <string>&amp;Send</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QStackedWidget" name="stack">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="sendWidget">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(215, 215, 215);
</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4" rowstretch="2">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item row="0" column="1">
           <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,0,0,0,0,0">
            <property name="spacing">
             <number>10</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,2">
              <property name="spacing">
               <number>20</number>
              </property>
              <item>
               <widget class="QLabel" name="sendKeyLabel">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>150</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Manjari</family>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>Enter the shared key of the exchange:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="sendKeyEdit">
                <property name="styleSheet">
                 <string notr="true">background-color: white;</string>
                </property>
                <property name="echoMode">
                 <enum>QLineEdit::Password</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="DropArea" name="dropArea" native="true">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>170</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
               </font>
              </property>
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="styleSheet">
               <string notr="true">#dropArea{border-radius: 10px;border:solid;border-width:2px;border-color:grey;background-color: rgb(186, 189, 182);}
</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="browseButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>500</width>
                <height>35</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>&amp;Browse</string>
              </property>
              <property name="icon">
               <iconset resource="resources.qrc">
                <normaloff>:/icons/document-open</normaloff>:/icons/document-open</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QProgressBar" name="sendProgress">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>50</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>14</pointsize>
                <italic>false</italic>
               </font>
              </property>
              <property name="value">
               <number>0</number>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="clearSendButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>500</width>
                <height>35</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="text">
               <string>&amp;Clear</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="sendButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>500</width>
                <height>35</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>&amp;Send</string>
              </property>
              <property name="autoDefault">
               <bool>false</bool>
              </property>
              <property name="default">
               <bool>false</bool>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="receiveWidget">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(215, 215, 215);
</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2" rowstretch="2,4" rowminimumheight="100,0">
          <property name="leftMargin">
           <number>20</number>
          </property>
          <property name="rightMargin">
           <number>20</number>
          </property>
          <item row="1" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0">
            <property name="spacing">
             <number>10</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="receiveLabel">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>500</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>15</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Enter a key and click Receive</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QProgressBar" name="receiveProgress">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
                <italic>false</italic>
               </font>
              </property>
              <property name="value">
               <number>0</number>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="clearReceiveButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimumSize">
               <size>
                <width>500</width>
                <height>35</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>&amp;Clear</string>
              </property>
             </widget>
            </item>
            <item alignment="Qt::AlignHCenter">
             <widget class="QPushButton" name="receiveButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>500</width>
                <height>35</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>&amp;Receive</string>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,2">
            <property name="spacing">
             <number>20</number>
            </property>
            <item>
             <widget class="QLabel" name="receiveKeyLabel">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>150</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Manjari</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Enter the shared key of the exchange:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="receiveKeyEdit">
              <property name="styleSheet">
               <string notr="true">background-color: white;</string>
              </property>
              <property name="echoMode">
               <enum>QLineEdit::PasswordEchoOnEdit</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="startWidget">
         <property name="styleSheet">
          <string notr="true">background-color: white;
border-style: solid;
border-radius: 10px;</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="logo">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">border-radius: 10px;</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="resources.qrc">:/logo/logo</pixmap>
            </property>
            <property name="scaledContents">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>841</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_Open"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_About">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menu_Options">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="action_Logs"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menu_Options"/>
   <addaction name="menu_About"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/document-open</normaloff>:/icons/document-open</iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="action_Logs">
   <property name="text">
    <string>&amp;Logs...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DropArea</class>
   <extends>QWidget</extends>
   <header>droparea.h</header>
   <container>1</container>
   <slots>
    <signal>changed()</signal>
    <slot>clear()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>receiveModeButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>switchToReceive()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>214</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendModeButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>switchToSend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>625</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>receiveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>waitForFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>285</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>retrieveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearSendButton</sender>
   <signal>clicked()</signal>
   <receiver>dropArea</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearSendButton</sender>
   <signal>clicked()</signal>
   <receiver>statusbar</receiver>
   <slot>clearMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>287</x>
     <y>548</y>
    </hint>
    <hint type="destinationlabel">
     <x>420</x>
     <y>642</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Open</sender>
   <signal>triggered()</signal>
   <receiver>browseButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>switchToReceive()</slot>
  <slot>switchToSend()</slot>
  <slot>retrieveFile()</slot>
  <slot>waitForFile()</slot>
  <slot>updateFormatsTable()</slot>
 </slots>
</ui>
